octoprint_filemanager/ThreadPool.py

```diff
@@ -1,33 +1,51 @@
-# http://code.activestate.com/recipes/577187-python-thread-pool/
+import sys
+IS_PY2 = sys.version_info < (3, 0)
+
+if IS_PY2:
+    from Queue import Queue
+else:
+    from queue import Queue
 
-from Queue import Queue
 from threading import Thread
 
+
 class Worker(Thread):
-    """Thread executing tasks from a given tasks queue"""
+    """ Thread executing tasks from a given tasks queue """
     def __init__(self, tasks):
         Thread.__init__(self)
         self.tasks = tasks
         self.daemon = True
         self.start()
-    
+
     def run(self):
         while True:
             func, args, kargs = self.tasks.get()
-            try: func(*args, **kargs)
-            except Exception as e: print(e)
-            self.tasks.task_done()
+            try:
+                func(*args, **kargs)
+            except Exception as e:
+                # An exception happened in this thread
+                print(e)
+            finally:
+                # Mark this task as done, whether an exception happened or not
+                self.tasks.task_done()
+
 
 class ThreadPool:
-    """Pool of threads consuming tasks from a queue"""
+    """ Pool of threads consuming tasks from a queue """
     def __init__(self, num_threads):
         self.tasks = Queue(num_threads)
-        for _ in range(num_threads): Worker(self.tasks)
+        for _ in range(num_threads):
+            Worker(self.tasks)
 
     def add_task(self, func, *args, **kargs):
-        """Add a task to the queue"""
+        """ Add a task to the queue """
         self.tasks.put((func, args, kargs))
 
+    def map(self, func, args_list):
+        """ Add a list of tasks to the queue """
+        for args in args_list:
+            self.add_task(func, args)
+
```

Comment:
Out of curiosity did you intend to use that function anywhere?
I couldn't find a reference to it.